### PR TITLE
EVG-14567 skip notifications patch isReady check for non-patches

### PR DESCRIPTION
**Pull request by ablack12**
[EVG-14567 ](https://jira.mongodb.org/browse/EVG-14567 )

### Description 
Periodic build notifications aren't sending, because non-patches (waterfall builds, periodic builds, git tag versions) don't have patches, so isReady is being set to false.

### Testing 
Not tested; reasoned over.

### CODE REVIEW

File: trigger/version.go [hunk 1]
```diff
@@ -293,14 +293,18 @@ func (t *versionTriggers) versionRegression(sub *event.Subscription) (*notificat
 	return nil, nil
 }
 
-func (t *versionTriggers) waitOnChildrenOrSiblings(sub *event.Subscription) (bool, error) {
+func (t *versionTriggers) waitOnChildrenOrSiblings() (bool, error) {
```
Comment (ablack12) [inline trigger/version.go:296]: removed sub since it's not doing anything

File: trigger/version.go [hunk 2]
```diff
@@ -293,14 +293,18 @@ func (t *versionTriggers) versionRegression(sub *event.Subscription) (*notificat
...
+	// only patches have to wait on children/siblings
+	if !evergreen.IsPatchRequester(t.version.Requester) {
+		return true, nil
+	}
 	isReady := false
 	patchDoc, err := patch.FindOne(patch.ByVersion(t.version.Id))
 	if err != nil {
-		return isReady, errors.Wrapf(err, "error getting patchDoc '%s'", t.version.Id)
+		return isReady, errors.Wrapf(err, "error getting patch '%s'", t.version.Id)
```
Comment (ablack12) [inline trigger/version.go:304]: sorry for the unrelated change; it's best for errors to say what the error actually is rather than using variable names

**Review by malikchaya2 — APPROVED** [review]
